crates/script/src/simulate.rs

```diff
@@ -113,9 +113,10 @@ impl PreSimulationState {
         // Executes all transactions from the different forks concurrently.
         let futs = transactions
             .into_iter()
-            .map(|transaction| async {
+            .map(|mut transaction| async {
                 let mut runner = runners.get(&transaction.rpc).expect("invalid rpc url").write();
 
+                let zk_metadata = transaction.zk.clone();
```

Comment:
I think also because `transaction` is later `&mut` borrowed (at `tx_mut`) so we can't get the field by itself

Reply:
`tx_mut()` takes a mutable reference to transaction but returns another type that does not have zk fields, you cannot get it from `transaction` because `tx` is holding the reference